servo/connectors/kubernetes.py

```diff
@@ -2463,9 +2482,13 @@ async def apply(self) -> None:
         # The resource_version attribute lets us efficiently watch for changes
         # reference: https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes
         """
-        async with self.deployment.rollout(timeout=self.timeout) as deployment:
-            # Patch the Deployment via the Kubernetes API
-            await deployment.patch()
+        try:
+            async with self.deployment.rollout(timeout=self.timeout) as deployment:
+                # Patch the Deployment via the Kubernetes API
+                await deployment.patch()
+        except WatchTimeoutError:
+            servo.logger.error(f"Timed out waiting for Deployment to become ready...")
+            await self.raise_for_status()
```

Comment:
What's the functional difference here versus the previous raise_for_status in the patch code path? What are we gaining?

Reply:
The previous raise_for_status in the patch code path does not have access to the adjustments applied to the optimization (nor should it) because it is being called by the Deployment class instance. Adding WatchTimeoutError brings the implementation more in line with that of the tuning optimization's TimeoutError catch block and allows the optimization object to pass the current adjustments into raise_for_status:
https://github.com/opsani/servox/blob/42c20e3dc5a1e466ff81ee971c5f9be34d96b9ec/servo/connectors/kubernetes.py#L2504-L2506